week-8-practise/1-code-refactoring.js

```diff
@@ -8,28 +8,28 @@
  * There may be both syntax and logical errors present.
  */
 
-var reversed;
-let capitilised;
 
-// Function that reverses and capitlises text
-exports function reverseAndCaptilise(text) 
-  var capitlised = text.toUpper()
-  reversed = capitilised.split('').reverse().join('') 
-
-  return reversed
-  }
+// this function handle the split the reverse ad the join method
+export function reverseAndCaptilise(text) {
+  return text
+    .split("")
+    .reverse()
+    .join("")
+    .toUpperCase();
+}
 
-// Function that returns text with every other letter capitlised
+//this function will transform all text to upperCase,
 export function everyOtherLetterCapitilised(text) {
-  for (i==0; i++; i<text.length) {
-      if (i/2==0) {
-       text[i] = text[i].toUpperCase() 
-    }    
-  } 
+  return text.toUpperCase();
```

Comment:
The intention of the function is to capitilise every other letter (i.e. "practise" becomes "PrAcTiSe"). With that, the for loop should stay, though there is a logical error in the for loop.

Reply:
@sansaid  As the instruction was not too clear on the intention of the function, I thought I might as well just capitalise all the string, I have re-inserted the loop in the function, will appreciate if you review it again

Reply:
No worries - I'll take responsibility for that, there was definitely scope to make it clearer. Thanks!

Reply:
@sansaid  what do you think about how I used the loop in this exercise?..., was that your expectation?